agent/api/container.go

```diff
@@ -95,6 +107,9 @@ type Container struct {
 	// 'Internal' containers are ones that are not directly specified by
 	// task definitions, but created by the agent
 	IsInternal bool
+	// InternalContainerType represents the internal container type when
+	// IsInternal is set to true
+	InternalContainerType InternalContainer
```

Comment:
I was thinking if we can get rid of the `IsInternal` field by adding another enum type `InternalContainerNone`, but not sure if this is a good idea?